app/services/invoices/regenerate_from_voided_service.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+module Invoices
+  class RegenerateFromVoidedService < BaseService
+    Result = BaseResult[:invoice]
+
+    def initialize(voided_invoice:, fees_params:)
+      @voided_invoice = voided_invoice
+      @fees_params = fees_params
+      @regenerated_invoice = nil
+      super
+    end
+
+    activity_loggable(
+      action: "invoice.regenerated",
+      record: -> { voided_invoice }
+    )
+
+    def call
+      return result.not_found_failure!(resource: "invoice") unless voided_invoice
+
+      ActiveRecord::Base.transaction do
```

Comment:
We are calling multiple sub-services, don't we want to test the result or at least to use `call!` to raise the failed results?